flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/ConsumerVertexGroup.java

```diff
@@ -66,4 +75,12 @@ public boolean isEmpty() {
     public ExecutionVertexID getFirst() {
         return iterator().next();
     }
+
+    public ConsumedPartitionGroup getConsumedPartitionGroup() {
+        return checkNotNull(consumedPartitionGroup, "ConsumedPartitionGroup is not properly set.");
+    }
+
+    public void setConsumedPartitionGroup(ConsumedPartitionGroup consumedPartitionGroup) {
```

Comment:
Maybe add check `this.consumedPartitionGroup == null || this.consumedPartitionGroup == consumedPartitionGroup` here

Reply:
`this.consumedPartitionGroup == null` is good enough I think. We do not expect this set method of one group to be invoked multiple times.